vms/proposervm/block/block.go

```diff
@@ -40,6 +45,20 @@ type SignedBlock interface {
 	// Proposer returns the ID of the node that proposed this block. If no node
 	// signed this block, [ids.EmptyNodeID] will be returned.
 	Proposer() ids.NodeID
+
+	VRFSig() []byte
+
+	// VerifySignature validates the correctness of the VRF signature.
+	VerifySignature(*bls.PublicKey, []byte, ids.ID, uint32) bool
+}
+
+// statelessUnsignedBlockV0 is the "old" version of statelessUnsignedBlock, which doesn't contain the VRF signature.
+type statelessUnsignedBlockV0 struct {
+	ParentID     ids.ID `serialize:"true"`
+	Timestamp    int64  `serialize:"true"`
+	PChainHeight uint64 `serialize:"true"`
+	Certificate  []byte `serialize:"true"`
+	Block        []byte `serialize:"true"`
```

Comment:
I think a better pattern would be to bump the codec version instead of two separate structs

We do this in the validator metadata here: https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/state/metadata_codec.go

and how it is used is here: https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/state/metadata_validator.go#L31-L40

where the `StakerStartTime` is only serialized if codec version 1 is used

Reply:
This would also remove the `panic` calls later on in the generic version

Reply:
thats very interesting. I wasn't aware of the conditional serialization. I will take another attempt with that.